src/main/java/life/qbic/portal/views/AppMainView.java

```diff
@@ -18,15 +18,17 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
 @PageTitle("Subscription Cancellation")
 @Route(value = "cancel")
 @RouteAlias(value = "")
 public class AppMainView extends HorizontalLayout implements HasUrlParameter<String> {
 
   private final CancelSubscriptionInput cancelSubscriptionInput;
-  Logger logger = LoggerFactory.getLogger(this.getClass());
+  private static final Logger logger = LoggerFactory.getLogger(AppMainView.class);
 
+  //do not add the component annotation to the class as suggested, it causes the app to fail when loading multiple requests
```

Comment:
Why is this the case? Why does it fail? Can you provide some more context here?

Reply:
When I load the app via an HTTP request it works for the first request and fails for the second with an internal server error because of some IllegalStateException. It seems as if there are duplicate nodes or so --> something spring related. I think this can be fixed in the future, if the easy fix is to just remove the component annotation here. But I am also happy to get some more insights here from someone more familiar with spring than me.